control/controlclient/direct.go

```diff
@@ -133,6 +135,20 @@ type Options struct {
 	// MapResponse.PingRequest queries from the control plane.
 	// If nil, PingRequest queries are not answered.
 	Pinger Pinger
+
+	// DialPlan contains and stores a previous dial plan that we received
+	// from the control server; if nil, we fall back to using DNS.
+	//
+	// If we receive a new DialPlan from the server, this value will be
+	// updated.
+	DialPlan ControlDialPlanner
+}
+
+// ControlDialPlanner is implemented by an atomic.Pointer to
```

Comment:
```
// ControlDialPlanner is the interface optionally supplied when creating a control client
// to control exactly how TCP connections to the control plane are dialed.
//
// It is usually implemented by an atomic.Pointer.
type .. {
    // Load returns the current plan for how to connect to control.
    //
    // It can (or can not?) be nil. If so that means ...

    // Store updates the dial plan from new directions from the control server.
    // The dial plan can span multiple connections to the control plan.
    // That is, an dial plan received when connected over wi-fi is still valid for
    // a subsequent connection over LTE after a network switch.
```

Reply:
Used this with minor tweaks :+1: